### PR TITLE
Spellcheck language settings (enable/disable & dictionary languages)

### LINKED ISSUE #2183: SETTINGS: Spell-check: Languages
**Did you search for similar issues before submitting this one?**
Yes, terms: `spell check` `spell-check` `language`
**Describe the issue you encountered:**
Is it possible to change the spell-check language, similar to chrome?
**Expected behavior:**
Right click --> set language ...

### PR DESCRIPTION
**Pull request by darkdh**
fix #2183

Auditors: @bbondy, @bsclifton

Submitter Checklist:

- [x] Submitted a [ticket](https://github.com/brave/browser-laptop/issues) for my issue if one did not already exist.
- [x] Used Github [auto-closing keywords](https://help.github.com/articles/closing-issues-via-commit-messages/) in the commit message.
- [x] Added/updated tests for this change (for new code or code which already has tests).
- [x] Ran `git rebase -i` to squash commits (if needed).
- [x] Tagged reviewers and labelled the pull request [as needed](https://github.com/brave/browser-laptop/wiki/Pull-request-process).

Test Plan:

Reviewer Checklist:

Tests

- [ ] Adequate test coverage exists to prevent regressions
- [ ] Tests should be independent and work correctly when run individually or as a suite [ref](https://github.com/brave/browser-laptop/wiki/Code-Guidelines#test-dependencies)
- [ ] New files have MPL2 license header

### CODE REVIEW

File: app/browser/reducers/spellCheckerReducer.js [hunk 1]
```diff
@@ -28,12 +31,29 @@ const migrate = (state) => {
...
+}
+
 const spellCheckerReducer = (state, action, immutableAction) => {
   action = immutableAction || makeImmutable(action)
   switch (action.get('actionType')) {
     case appConstants.APP_SET_STATE:
       state = migrate(state)
       break
+    case appConstants.APP_CHANGE_SETTING:
+      // TODO(darkdh): APP_SET_STATE is too early for startup setting so we use
```
Comment (bridiver) [inline app/browser/reducers/spellCheckerReducer.js]: why is it too early?

Comment (darkdh) [reply app/browser/reducers/spellCheckerReducer.js]: I don't know the root cause but when I tried to use APP_SET_STATE, it has no effect for these two preferences.

**Comment by darkdh** [comment]
unit test fixed. waiting for approval from @bsclifton

**Comment by bbondy** [comment]
master: fc757e329bed659917f9972ddf2f321d450eefe2
0.21.x: 603d45a72d468ea9664b611e1c40cbd0ea44530a
0.20.x: 92af07a1bc86857ddface299740c0492c0d07e79

0.19.x:
4b258b09d639d14f667c68438cb1bf58ae8dbd7f
Also had to pull in this type test fix: ba3b08f9f67b163055cb9f48295c07d8a8acfd86

File: js/about/preferences.js [hunk 2]
```diff
@@ -11,8 +11,6 @@ const {StyleSheet, css} = require('aphrodite/no-important')
 const globalStyles = require('../../app/renderer/components/styles/global')
 const commonStyles = require('../../app/renderer/components/styles/commonStyles')
 const locale = require('../../js/l10n')
-// TODO(cezaraugusto): replace this with our own style
```
Comment (luixxiul) [inline js/about/preferences.js]: @cezaraugusto Was this abandoned? :-p

**Comment by darkdh** [comment]
extra change for 0.19.x: 8453020a67e3d23ef8722af492668e0c773de5ca

**Pull request merged**